homework01/rudolf/vector.cpp

```diff
@@ -0,0 +1,46 @@
+#include "vector.h"
+
+Vector::Vector()
+{
+	mX=0;
+	mY=0;
+}
+
+Vector::Vector(double x, double y): mX(x), mY(y){}
+
+double Vector::dotProduct(const Vector& vec) const
+{
+	return mX * vec.mX + mY * vec.mY ;
+}
+
+double Vector::GetX() const
+{
+	return mX;
```

Comment:
We would usually define those kind of methods (getters, one liners) in the header file, but it's not wrong to do it like that.